src/_cffi_src/openssl/dh.py

```diff
@@ -9,7 +9,20 @@
 """
 
 TYPES = """
-typedef ... DH;
+typedef struct dh_st {
+    /* Prime number (shared) */
+    BIGNUM *p;
+    /* Generator of Z_p (shared) */
+    BIGNUM *g;
+    /* Private DH value x */
+    BIGNUM *priv_key;
+    /* Public DH value g^x */
+    BIGNUM *pub_key;
+    /* X9.42/RFC 2631 */
+    BIGNUM *q;
+    BIGNUM *j;
+    ...;
+} DH;
```

Comment:
mmm, so this definitiion, using the struct fields, won't work in OpenSSL 1.1.0, so this will have to be ported to use the various accessors that OpenSSL exposes.